python/pip_install/tools/dependency_resolver/dependency_resolver.py

```diff
@@ -128,6 +131,8 @@ def _locate(bazel_runfiles, file):
     os.environ["LC_ALL"] = "C.UTF-8"
     os.environ["LANG"] = "C.UTF-8"
 
+    argv = []
```

Comment:
It looks like we lose sys.argv[0] compared to before. I'm not sure if the piptools cli expects argv to have argv[0] be the name of the original script but maybe we can add sys.argv[0] here in case.

Reply:
Just to be clear, are you suggesting we change this to the following?

```suggestion
    argv = [sys.argv[0]]
```

Reply:
From what I can tell, omitting argv[0] when calling cli() is correct.

Under the hood, piptools is using click, and the implementation is obscured by a ton of decorators. Digging into it...
* `compile.cli` is an instance of the click `Command` class.
* `Command.__call__` is an alias for `Command.main`
* `Command.main` docs say, for the first positionl arg (named `args`): "the arguments that should be used for parsing. If not provided sys.argv[1:] is used"

So it doesn't expect args to contain argv[0]. The docs go on further to say the optional `prog_name` arg is initialized to sys.argv[0] if not specified.

As a side note, there is a `standalone_mode` arg, which controls whether the command tries to exit or not.